tools/lsp/server_loop.rs

```diff
@@ -483,64 +458,81 @@ pub async fn set_binding_command(
-            };
-
-            if start != element_range.start {
-                return Err(format!(
-                    "Element found, but does not start at the expected place (){start:?} != {:?}).",
-                    element_range.start
-                )
-                .into());
-            }
-            if end != element_range.end {
-                return Err(format!(
-                    "Element found, but does not end at the expected place (){end:?} != {:?}).",
-                    element_range.end
-                )
-                .into());
-            }
-
-            let (result, edit) = properties::set_binding(
-                document_cache,
-                &uri,
-                &element,
-                &property_name,
-                new_expression,
-            )?;
-
-            if !dry_run {
-                if let Some(edit) = edit {
-                    let response = notifier
-                        .send_request::<lsp_types::request::ApplyWorkspaceEdit>(
-                            lsp_types::ApplyWorkspaceEditParams {
-                                label: Some("set binding".into()),
-                                edit,
-                            },
-                        )?
-                        .await?;
-                    if !response.applied {
-                        return Err(response
-                            .failure_reason
-                            .unwrap_or("Operation failed, no specific reason given".into())
-                            .into());
-                    }
+    let (result, edit) = {
+        let document_cache = &mut ctx.document_cache.borrow_mut();
+        let uri = text_document.uri;
+        if let Some(source_version) = text_document.version {
+            if let Some(current_version) = document_cache.document_version(&uri) {
+                if current_version != source_version {
+                    return Err(
+                        "Document version mismatch. Please refresh your property information"
+                            .into(),
+                    );
                 }
+            } else {
+                return Err(
+                    format!("Document with uri {} not found in cache", uri.to_string()).into()
+                );
             }
+        }
 
-            Ok(serde_json::to_value(result).expect("Failed to serialize set_binding result!"))
-        } else {
-            Err("The element was found, but had no range defined!".into()).into()
+        let element = element_at_position(document_cache, &uri, &element_range.start).ok_or_else(
+            || -> Error {
+                format!("No element found at the given start position {:?}", &element_range.start)
+                    .into()
+            },
+        )?;
+        let node_range = element
+            .borrow()
+            .node
+            .as_ref()
+            .ok_or_else(|| -> Error { "The element was found, but had no range defined!".into() })?
```

Comment:
```suggestion
            .ok_or_else(|| "The element was found, but had no range defined!")?
```

I just wonder why the type is requited.  I think the `?` already have an Into builtin.
This is just a style issue anyway, ignore that if that doesn't work.